grafana/resource_dashboard.go

```diff
@@ -188,33 +189,48 @@ func ReadDashboard(ctx context.Context, d *schema.ResourceData, meta interface{}
 		}
 	}
 
+	d.SetId(dashboard.Model["uid"].(string))
+	d.Set("uid", dashboard.Model["uid"].(string))
+	d.Set("slug", dashboard.Meta.Slug)
+	d.Set("folder", dashboard.Folder)
+	d.Set("dashboard_id", int64(dashboard.Model["id"].(float64)))
+	d.Set("version", int64(dashboard.Model["version"].(float64)))
+
 	configJSONBytes, err := json.Marshal(dashboard.Model)
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	remoteDashJSON := unmarshalDashboardConfigJSON(string(configJSONBytes))
 
-	configJSON := normalizeDashboardConfigJSON(string(configJSONBytes))
+	// If `uid` is not set in configuration, we need to delete it from the
+	// dashboard JSON we just read from the Grafana API. This is so it does not
+	// create a diff. We can assume the uid was randomly generated by Grafana or
+	// it was removed after dashboard creation. In any case, the user doesn't
+	// care to manage it.
+	if d.Get("config_json").(string) != "" {
+		configuredDashJSON := unmarshalDashboardConfigJSON(d.Get("config_json").(string))
```

Comment:
How about storing the result of casting the value into a var, like:
```suggestion
	if configJson := d.Get("config_json").(string); configJson != "" {
		configuredDashJSON := unmarshalDashboardConfigJSON(configJson)
```